src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -370,15 +370,17 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
     * @return Date
     */ 
     private static Date dtStartOrEndOfAddrSeasonal(Address__c addr, boolean fStart) {    
+        integer startYear = integer.valueOf(addr.Seasonal_Start_Year__c);
```

Comment:
Regarding the first comment, I tested the logic in the Anonymous window of the debug console with the following code, and no exception is thrown:

Decimal test = null;
integer startYear = integer.valueOf(test);
System.debug('Start year: ' + startYear);

Which makes sense (though Apex sometimes surprises you), since the Integer value of null should just be null.